src/LinuxTracing/UprobesUnwindingVisitorTest.cpp

```diff
@@ -0,0 +1,257 @@
+
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kTargetMapsStart), Lt(kTargetMapsEnd))))
+        .WillRepeatedly(Return(&kTargetMapInfo));
+
+    EXPECT_CALL(maps_, Find(AllOf(Ge(kNonExecutableMapsStart), Lt(kNonExecutableMapsEnd))))
+        .WillRepeatedly(Return(&non_executable_map_info_));
+  }
+
+  void TearDown() override { visitor_.reset(); }
+
+  MockLibunwindstackMaps maps_;
+  MockLibunwindstackUnwinder unwinder_;
+  MockTracerListener listener_;
+
+  std::unique_ptr<UprobesUnwindingVisitor> visitor_ = nullptr;
+
+  static constexpr uint64_t kUprobesMapsStart = 42;
+  static constexpr uint64_t kUprobesMapsEnd = 84;
+
+  static constexpr uint64_t kTargetMapsStart = 100;
+  static constexpr uint64_t kTargetMapsEnd = 200;
+
+  static constexpr uint64_t kNonExecutableMapsStart = 500;
+  static constexpr uint64_t kNonExecutableMapsEnd = 600;
+
+  static constexpr uint64_t kTargetAddress1 = 100;
+  //  static constexpr uint64_t kUprobesAddress = 42;
+  static constexpr uint64_t kTargetAddress2 = 200;
+  static constexpr uint64_t kTargetAddress3 = 300;
+  //  static constexpr uint64_t kNonExecutableAddress3 = 500;
+
+  static inline const std::string kUprobesName = "[uprobes]";
+  static inline const std::string kTargetName = "target";
+  static inline const std::string kNonExecutableName = "data";
+
+  static inline unwindstack::MapInfo kUprobesMapInfo{
+      nullptr, kUprobesMapsStart, kUprobesMapsEnd, 0, PROT_EXEC | PROT_READ, kUprobesName};
+
+  static inline unwindstack::MapInfo kTargetMapInfo{nullptr, kTargetMapsStart,      kTargetMapsEnd,
+                                                    0,       PROT_EXEC | PROT_READ, kTargetName};
+
+  static inline unwindstack::MapInfo non_executable_map_info_{
+      nullptr, kNonExecutableMapsStart, kNonExecutableMapsEnd,
+      0,       PROT_EXEC | PROT_READ,   kNonExecutableName};
+};
+
+}  // namespace
+
+TEST_F(UprobesUnwindingVisitorTest, VisitValidStackSampleWithoutUprobesEmitsEvents) {
+  constexpr uint32_t kPid = 10;
+  constexpr uint64_t kStackSize = 13;
+  StackSamplePerfEvent event{kStackSize};
+  perf_event_sample_id_tid_time_streamid_cpu sample_id{
+      .pid = kPid,
+      .tid = 11,
+      .time = 15,
+      .stream_id = 12,
+      .cpu = 0,
+      .res = 0,
+  };
+  event.ring_buffer_record->sample_id = sample_id;
+
+  EXPECT_CALL(maps_, Get).Times(1).WillOnce(Return(nullptr));
+
+  std::vector<unwindstack::FrameData> libunwindstack_callstack;
+
+  unwindstack::FrameData frame_1{
+      .pc = kTargetAddress1, .function_name = "foo", .function_offset = 0, .map_name = kTargetName};
+  libunwindstack_callstack.push_back(frame_1);
+  unwindstack::FrameData frame_2{
+      .pc = kTargetAddress2, .function_name = "bar", .function_offset = 0, .map_name = kTargetName};
+  libunwindstack_callstack.push_back(frame_2);
+  unwindstack::FrameData frame_3{
+      .pc = kTargetAddress3, .function_name = "baz", .function_offset = 0, .map_name = kTargetName};
```

Comment:
Then I'd suggest the trailing comma also here...

Reply:
Done.